core/chains/cosmos/cosmostxm/txm_test.go

```diff
@@ -0,0 +1,131 @@
+//go:build integration
+
+package cosmostxm_test
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/onsi/gomega"
+	uuid "github.com/satori/go.uuid"
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/require"
+
+	cosmosclient "github.com/smartcontractkit/chainlink-terra/pkg/cosmos/client"
+	coscfg "github.com/smartcontractkit/chainlink-terra/pkg/cosmos/config"
+
+	"github.com/smartcontractkit/chainlink/core/chains/cosmos"
+	"github.com/smartcontractkit/chainlink/core/chains/cosmos/cosmostxm"
+	"github.com/smartcontractkit/chainlink/core/internal/cltest/heavyweight"
+	"github.com/smartcontractkit/chainlink/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/core/internal/testutils/pgtest"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/chainlink"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/core/services/pg"
+	"github.com/smartcontractkit/chainlink/core/utils"
+
+	. "github.com/smartcontractkit/chainlink-terra/pkg/cosmos/db"
+)
+
+func TestTxm_Integration(t *testing.T) {
+	t.Skip("requires cosmosd")
+	chainID := fmt.Sprintf("Chainlinktest-%d", rand.Int31n(999999))
+	fallbackGasPrice := sdk.NewDecCoinFromDec("uatom", sdk.MustNewDecFromStr("0.01"))
+	chain := cosmos.CosmosConfig{ChainID: &chainID, Enabled: ptr(true), Chain: coscfg.Chain{
+		FallbackGasPriceUAtom: ptr(decimal.RequireFromString("0.01")),
+		GasLimitMultiplier:    ptr(decimal.RequireFromString("1.5")),
+	}}
+	cfg, db := heavyweight.FullTestDBNoFixturesV2(t, "cosmos_txm", func(c *chainlink.Config, s *chainlink.Secrets) {
+		c.Cosmos = cosmos.CosmosConfigs{&chain}
+	})
+	lggr := logger.TestLogger(t)
+	logCfg := pgtest.NewQConfig(true)
+	gpe := cosmosclient.NewMustGasPriceEstimator([]cosmosclient.GasPricesEstimator{
+		cosmosclient.NewFixedGasPriceEstimator(map[string]sdk.DecCoin{
+			"uatom": fallbackGasPrice,
+		}),
+	}, lggr)
+	orm := cosmostxm.NewORM(chainID, db, lggr, logCfg)
+	eb := pg.NewEventBroadcaster(cfg.DatabaseURL(), 0, 0, lggr, uuid.NewV4())
+	require.NoError(t, eb.Start(testutils.Context(t)))
+	t.Cleanup(func() { require.NoError(t, eb.Close()) })
+	ks := keystore.New(db, utils.FastScryptParams, lggr, pgtest.NewQConfig(true))
+	accounts, testdir, tendermintURL := cosmosclient.SetupLocalCosmosNode(t, "42")
```

Comment:
This test still has a bunch of issues but it's an integration test that's usually disabled.